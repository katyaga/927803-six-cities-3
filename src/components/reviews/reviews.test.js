import React from "react";
import renderer from "react-test-renderer";
import Reviews from "./reviews.jsx";

const MockComments = [
  {
    id: 10,
    authorAvatar: `img/avatar-angelina.jpg`,
    authorName: `Sara`,
    rating: 1,
    date: new Date(`2019-12-23 12:00:11`),
    text: `text1`,
  },
  {
    id: 20,
    authorAvatar: `img/avatar-max.jpg`,
    authorName: `Bob`,
    rating: 3,
    date: new Date(`2020-02-23 12:00:11`),
    text: `text2`,
  },
  {
    id: 30,
    authorAvatar: `img/avatar-max.jpg`,
    authorName: `Bob`,
    rating: 4,
    date: new Date(`2020-01-23 12:00:11`),
    text: `text3`,
  },
];

it(`Should Reviews render correctly`, () => {
  const tree = renderer
    .create(<Reviews
      comments={MockComments}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
