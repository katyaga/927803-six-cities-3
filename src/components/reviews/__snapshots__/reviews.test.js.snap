// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Reviews render correctly 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Bob
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        text2
      </p>
      <time
        className="reviews__time"
        dateTime="2020-02-23"
      >
        February 23, 2020
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Bob
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        text3
      </p>
      <time
        className="reviews__time"
        dateTime="2020-01-23"
      >
        January 23, 2020
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-angelina.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Sara
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "20%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        text1
      </p>
      <time
        className="reviews__time"
        dateTime="2019-12-23"
      >
        December 23, 2019
      </time>
    </div>
  </li>
</ul>
`;
